FWCore/Framework/interface/DataProxyTemplate.h

```diff
@@ -29,37 +39,31 @@
 // forward declarations
 
 namespace edm {
+
+  class EventSetupImpl;
+
   namespace eventsetup {
+
     template <class RecordT, class DataT>
     class DataProxyTemplate : public DataProxy {
     public:
       typedef DataT value_type;
       typedef RecordT record_type;
 
       DataProxyTemplate() {}
-      //virtual ~DataProxyTemplate();
 
-      // ---------- const member functions ---------------------
-
-      // ---------- static member functions --------------------
-
-      // ---------- member functions ---------------------------
-      const void* getImpl(const EventSetupRecordImpl& iRecord, const DataKey& iKey) override {
+      const void* getImpl(const EventSetupRecordImpl& iRecord,
+                          const DataKey& iKey,
+                          EventSetupImpl const* iEventSetupImpl) override {
         assert(iRecord.key() == RecordT::keyForClass());
         RecordT rec;
         rec.setImpl(&iRecord, std::numeric_limits<unsigned int>::max(), nullptr);
+        rec.setEventSetupImpl(iEventSetupImpl);
         return this->make(rec, iKey);
       }
 
     protected:
       virtual const DataT* make(const RecordT&, const DataKey&) = 0;
-
-    private:
-      DataProxyTemplate(const DataProxyTemplate&) = delete;  // stop default
```

Comment:
No. The base class (DataProxy) does not allow copy construction or assignment. Also this has no data members so there is no particular restriction coming from DataProxyTemplate that makes it not copyable or not assignable. I was just going for brevity. Do you think it would be better from the point of view of a human reading the code to add back in the lines that disallow copying and assigning?